### ISSUE TITLE
Negative consumption (Allegedly as returned) 

### ISSUE THREAD

**Issue by Churator** [GuyKh/iec-custom-component#72]
## System Information

version | core-2024.3.0
-- | --
installation_type | Home Assistant Container
dev | false
hassio | false
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.5.11-7-pve
arch | x86_64
timezone | Asia/Jerusalem
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1402
Downloaded Repositories | 15

</details>

<details><summary>AccuWeather</summary>

can_reach_server | ok
-- | --
remaining_requests | 37

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Dashboards</summary>

dashboards | 4
-- | --
resources | 8
views | 17
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | March 7, 2024 at 09:47
-- | --
current_recorder_run | March 13, 2024 at 23:38
estimated_db_size | 253.78 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/ludeeus/iec/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Once a day, I see negative value
![image](https://github.com/GuyKh/iec-custom-component/assets/62523824/de081e70-e4a8-49b4-ae29-ec9d6c3c1b5e)

Not sure what it means, but during the negative value, I see one time status = 1024

{
            'status': 1024,
            'date': '2024-03-13T06:45:00.000000',
            'value': 0.945
        }

### Reproduction steps

view in energy tab

### Debug logs

```text
Latest reading

2024-03-15 22:54:12.072 | DEBUG    | iec_api.commons:send_post_request:176 - HTTP POST Response: {'status': 0, 'futureConsumptionInfo': {'lastInvoiceDate': None, 'currentDate': '2024-03-14', 'futureConsumption': 77.867, 'totalImport': 10137.364, 'totalImportDate': '2024-03-14'}, 'fromDate': '2024-03-13', 'toDate': '2024-03-13', 'totalConsumptionForPeriod': 41.921, 'totalImportDateForPeriod': '2024-03-13', 'meterStartDate': '2023-09-21', 'totalImport': 10095.443, 'data': [{'status': 0, 'date': '2024-03-13T00:00:00.000000', 'value': 0.359}, {'status': 0, 'date': '2024-03-13T00:15:00.000000', 'value': 0.354}, {'status': 0, 'date': '2024-03-13T00:30:00.000000', 'value': 0.463}, {'status': 0, 'date': '2024-03-13T00:45:00.000000', 'value': 0.334}, {'status': 0, 'date': '2024-03-13T01:00:00.000000', 'value': 0.364}, {'status': 0, 'date': '2024-03-13T01:15:00.000000', 'value': 0.433}, {'status': 0, 'date': '2024-03-13T01:30:00.000000', 'value': 0.331}, {'status': 0, 'date': '2024-03-13T01:45:00.000000', 'value': 0.388}, {'status': 0, 'date': '2024-03-13T02:00:00.000000', 'value': 0.407}, {'status': 0, 'date': '2024-03-13T02:15:00.000000', 'value': 0.352}, {'status': 0, 'date': '2024-03-13T02:30:00.000000', 'value': 0.397}, {'status': 0, 'date': '2024-03-13T02:45:00.000000', 'value': 0.391}, {'status': 0, 'date': '2024-03-13T03:00:00.000000', 'value': 0.369}, {'status': 0, 'date': '2024-03-13T03:15:00.000000', 'value': 0.427}, {'status': 0, 'date': '2024-03-13T03:30:00.000000', 'value': 0.359}, {'status': 0, 'date': '2024-03-13T03:45:00.000000', 'value': 0.353}, {'status': 0, 'date': '2024-03-13T04:00:00.000000', 'value': 0.948}, {'status': 0, 'date': '2024-03-13T04:15:00.000000', 'value': 0.987}, {'status': 0, 'date': '2024-03-13T04:30:00.000000', 'value': 0.848}, {'status': 0, 'date': '2024-03-13T04:45:00.000000', 'value': 1.008}, {'status': 0, 'date': '2024-03-13T05:00:00.000000', 'value': 0.836}, {'status': 0, 'date': '2024-03-13T05:15:00.000000', 'value': 0.877}, {'status': 0, 'date': '2024-03-13T05:30:00.000000', 'value': 0.971}, {'status': 0, 'date': '2024-03-13T05:45:00.000000', 'value': 0.412}, {'status': 0, 'date': '2024-03-13T06:00:00.000000', 'value': 0.463}, {'status': 0, 'date': '2024-03-13T06:15:00.000000', 'value': 0.357}, {'status': 0, 'date': '2024-03-13T06:30:00.000000', 'value': 0.969}, {'status': 1024, 'date': '2024-03-13T06:45:00.000000', 'value': 0.945}, {'status': 0, 'date': '2024-03-13T07:00:00.000000', 'value': 0.833}, {'status': 0, 'date': '2024-03-13T07:15:00.000000', 'value': 0.909}, {'status': 0, 'date': '2024-03-13T07:30:00.000000', 'value': 0.844}, {'status': 0, 'date': '2024-03-13T07:45:00.000000', 'value': 0.732}, {'status': 0, 'date': '2024-03-13T08:00:00.000000', 'value': 0.355}, {'status': 0, 'date': '2024-03-13T08:15:00.000000', 'value': 0.319}, {'status': 0, 'date': '2024-03-13T08:30:00.000000', 'value': 0.343}, {'status': 0, 'date': '2024-03-13T08:45:00.000000', 'value': 0.308}, {'status': 0, 'date': '2024-03-13T09:00:00.000000', 'value': 0.326}, {'status': 0, 'date': '2024-03-13T09:15:00.000000', 'value': 0.315}, {'status': 0, 'date': '2024-03-13T09:30:00.000000', 'value': 0.309}, {'status': 0, 'date': '2024-03-13T09:45:00.000000', 'value': 0.26}, {'status': 0, 'date': '2024-03-13T10:00:00.000000', 'value': 0.178}, {'status': 0, 'date': '2024-03-13T10:15:00.000000', 'value': 0.196}, {'status': 0, 'date': '2024-03-13T10:30:00.000000', 'value': 0.176}, {'status': 0, 'date': '2024-03-13T10:45:00.000000', 'value': 0.175}, {'status': 0, 'date': '2024-03-13T11:00:00.000000', 'value': 0.193}, {'status': 0, 'date': '2024-03-13T11:15:00.000000', 'value': 0.175}, {'status': 0, 'date': '2024-03-13T11:30:00.000000', 'value': 0.217}, {'status': 0, 'date': '2024-03-13T11:45:00.000000', 'value': 0.234}, {'status': 0, 'date': '2024-03-13T12:00:00.000000', 'value': 0.218}, {'status': 0, 'date': '2024-03-13T12:15:00.000000', 'value': 0.213}, {'status': 0, 'date': '2024-03-13T12:30:00.000000', 'value': 0.264}, {'status': 0, 'date': '2024-03-13T12:45:00.000000', 'value': 0.212}, {'status': 0, 'date': '2024-03-13T13:00:00.000000', 'value': 0.211}, {'status': 0, 'date': '2024-03-13T13:15:00.000000', 'value': 0.211}, {'status': 0, 'date': '2024-03-13T13:30:00.000000', 'value': 0.183}, {'status': 0, 'date': '2024-03-13T13:45:00.000000', 'value': 0.2}, {'status': 0, 'date': '2024-03-13T14:00:00.000000', 'value': 0.198}, {'status': 0, 'date': '2024-03-13T14:15:00.000000', 'value': 0.183}, {'status': 0, 'date': '2024-03-13T14:30:00.000000', 'value': 0.225}, {'status': 0, 'date': '2024-03-13T14:45:00.000000', 'value': 0.221}, {'status': 0, 'date': '2024-03-13T15:00:00.000000', 'value': 0.206}, {'status': 0, 'date': '2024-03-13T15:15:00.000000', 'value': 0.214}, {'status': 0, 'date': '2024-03-13T15:30:00.000000', 'value': 0.189}, {'status': 0, 'date': '2024-03-13T15:45:00.000000', 'value': 0.192}, {'status': 0, 'date': '2024-03-13T16:00:00.000000', 'value': 0.177}, {'status': 0, 'date': '2024-03-13T16:15:00.000000', 'value': 0.233}, {'status': 0, 'date': '2024-03-13T16:30:00.000000', 'value': 0.25}, {'status': 0, 'date': '2024-03-13T16:45:00.000000', 'value': 0.705}, {'status': 0, 'date': '2024-03-13T17:00:00.000000', 'value': 0.535}, {'status': 0, 'date': '2024-03-13T17:15:00.000000', 'value': 0.407}, {'status': 0, 'date': '2024-03-13T17:30:00.000000', 'value': 0.751}, {'status': 0, 'date': '2024-03-13T17:45:00.000000', 'value': 0.711}, {'status': 0, 'date': '2024-03-13T18:00:00.000000', 'value': 0.537}, {'status': 0, 'date': '2024-03-13T18:15:00.000000', 'value': 0.815}, {'status': 0, 'date': '2024-03-13T18:30:00.000000', 'value': 1.01}, {'status': 0, 'date': '2024-03-13T18:45:00.000000', 'value': 0.523}, {'status': 0, 'date': '2024-03-13T19:00:00.000000', 'value': 0.318}, {'status': 0, 'date': '2024-03-13T19:15:00.000000', 'value': 0.269}, {'status': 0, 'date': '2024-03-13T19:30:00.000000', 'value': 0.26}, {'status': 0, 'date': '2024-03-13T19:45:00.000000', 'value': 0.255}, {'status': 0, 'date': '2024-03-13T20:00:00.000000', 'value': 0.261}, {'status': 0, 'date': '2024-03-13T20:15:00.000000', 'value': 0.232}, {'status': 0, 'date': '2024-03-13T20:30:00.000000', 'value': 0.236}, {'status': 0, 'date': '2024-03-13T20:45:00.000000', 'value': 0.232}, {'status': 0, 'date': '2024-03-13T21:00:00.000000', 'value': 0.33}, {'status': 0, 'date': '2024-03-13T21:15:00.000000', 'value': 0.284}, {'status': 0, 'date': '2024-03-13T21:30:00.000000', 'value': 0.306}, {'status': 0, 'date': '2024-03-13T21:45:00.000000', 'value': 0.296}, {'status': 0, 'date': '2024-03-13T22:00:00.000000', 'value': 0.59}, {'status': 0, 'date': '2024-03-13T22:15:00.000000', 'value': 1.148}, {'status': 0, 'date': '2024-03-13T22:30:00.000000', 'value': 0.853}, {'status': 0, 'date': '2024-03-13T22:45:00.000000', 'value': 0.773}, {'status': 0, 'date': '2024-03-13T23:00:00.000000', 'value': 0.639}, {'status': 0, 'date': '2024-03-13T23:15:00.000000', 'value': 0.585}, {'status': 0, 'date': '2024-03-13T23:30:00.000000', 'value': 0.525}, {'status': 0, 'date': '2024-03-13T23:45:00.000000', 'value': 0.436}]}
```

**Comment by GuyKh** [comment]
Looks very much like #71.

AFAIK 1024 maybe a partial hour read. One approach would be to filter it out from the LongTerm Statistics

**Comment by Churator** [comment]
> Looks very much like #71.
> 
> AFAIK 1024 maybe a partial hour read. One approach would be to filter it out from the LongTerm Statistics

how can I do that ? 
#71 has only select command

**Comment by GuyKh** [comment]
@Churator , not telling **you** to do it, more like thinking aloud.

What I think is that due to data return grouping by hour, the iteration over the keys isn't in order. So in #74 - I added a sorted iteration. 🤷🏻 if it will work.

**Comment by haimn** [comment]
I'm still getting negative values after the update:
![Screenshot_20240321_091217_Home Assistant](https://github.com/GuyKh/iec-custom-component/assets/1581319/0d367dfc-96b3-4354-b918-986755103a3a)

**Comment by GuyKh** [comment]
👍🏻

**Comment by GuyKh** [comment]
I'll check again

**Comment by haimn** [comment]
> 👍🏻 Reopened. Is your logs in debug (especially `iec-api`)? If so, can you share logs?

I'll remove sensitive data from the logs and I'll share them later.
Thank you.

**Comment by GuyKh** [comment]
```
24-03-29 03:46:41.121 DEBUG (MainThread) [custom_components.iec.coordinator] Last statistics are from 2024-03-27 23:00:00
2024-03-29 03:46:41.122 DEBUG (MainThread) [custom_components.iec.coordinator] Fetching consumption from 2024-03-28 01:00:00
2024-03-29 03:46:41.123 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iecapi.iec.co.il//api/Consumption/RemoteReadingRange/123456789
2024-03-29 03:46:41.123 DEBUG (MainThread) [iec_api.commons] HTTP Content: {'meterSerialNumber': '12345', 'meterCode': '12', 'lastInvoiceDate': '2024-03-28', 'fromDate': '2024-03-28', 'resolution': 1}
2024-03-29 03:46:42.515 DEBUG (MainThread) [iec_api.commons] HTTP POST Response: {'status': 0, 'futureConsumptionInfo': {'lastInvoiceDate': None, 'currentDate': '2024-03-28', 'futureConsumption': 0.122, 'totalImport': 12904.741, 'totalImportDate': '2024-03-28'}, 'fromDate': '2024-03-28', 'toDate': '2024-03-28', 'totalConsumptionForPeriod': 0.122, 'totalImportDateForPeriod': '2024-03-28', 'meterStartDate': '2023-03-17', 'totalImport': 12904.741, 'data': [{'status': 0, 'date': '2024-03-28T00:00:00.000000', 'value': 0.117}, {'status': 0, 'date': '2024-03-28T00:15:00.000000', 'value': 0.111}, {'status': 0, 'date': '2024-03-28T00:30:00.000000', 'value': 0.127}, {'status': 0, 'date': '2024-03-28T00:45:00.000000', 'value': 0.116}, {'status': 0, 'date': '2024-03-28T01:00:00.000000', 'value': 0.08}, {'status': 0, 'date': '2024-03-28T01:15:00.000000', 'value': 0.071}, {'status': 0, 'date': '2024-03-28T01:30:00.000000', 'value': 0.04}]}
2024-03-29 03:46:42.521 DEBUG (MainThread) [custom_components.iec.coordinator] Last Consumption Sum for 123456789: 1144.4980000000005
2024-03-29 03:46:42.521 DEBUG (MainThread) [custom_components.iec.coordinator] Last Estimated Cost Sum for 123456789: 703.2940209999996
2024-03-29 03:46:42.522 DEBUG (MainThread) [custom_components.iec.coordinator] Last hour fetched for 123456789: 2024-03-28 01:00:00+02:00
2024-03-29 03:46:42.522 DEBUG (MainThread) [custom_components.iec.coordinator] New Consumption Sum for 123456789: 1145.1600000000005
2024-03-29 03:46:42.522 DEBUG (MainThread) [custom_components.iec.coordinator] New Estimated Cost Sum for 123456789: 703.7008199999996

2024-03-30 13:42:38.678 DEBUG (MainThread) [custom_components.iec.coordinator] Last statistics are from 2024-03-28 01:00:00
2024-03-30 13:42:38.678 DEBUG (MainThread) [custom_components.iec.coordinator] Fetching consumption from 2024-03-28 01:00:00
2024-03-30 13:42:38.678 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iecapi.iec.co.il//api/Consumption/RemoteReadingRange/123456789
2024-03-30 13:42:38.678 DEBUG (MainThread) [iec_api.commons] HTTP Content: {'meterSerialNumber': '12345', 'meterCode': '12', 'lastInvoiceDate': '2024-03-28', 'fromDate': '2024-03-28', 'resolution': 1}
2024-03-30 13:42:40.060 DEBUG (MainThread) [iec_api.commons] HTTP POST Response: {'status': 0, 'futureConsumptionInfo': {'lastInvoiceDate': None, 'currentDate': '2024-03-28', 'futureConsumption': 0.122, 'totalImport': 12945.4, 'totalImportDate': '2024-03-30'}, 'fromDate': '2024-03-28', 'toDate': '2024-03-28', 'totalConsumptionForPeriod': 0.122, 'totalImportDateForPeriod': '2024-03-28', 'meterStartDate': '2023-03-17', 'totalImport': 12904.741, 'data': [{'status': 0, 'date': '2024-03-28T00:00:00.000000', 'value': 0.117}, {'status': 0, 'date': '2024-03-28T00:15:00.000000', 'value': 0.111}, {'status': 0, 'date': '2024-03-28T00:30:00.000000', 'value': 0.127}, {'status': 0, 'date': '2024-03-28T00:45:00.000000', 'value': 0.116}, {'status': 0, 'date': '2024-03-28T01:00:00.000000', 'value': 0.08}, {'status': 0, 'date': '2024-03-28T01:15:00.000000', 'value': 0.071}, {'status': 0, 'date': '2024-03-28T01:30:00.000000', 'value': 0.04}]}
2024-03-30 13:42:40.076 DEBUG (MainThread) [custom_components.iec.coordinator] Last Consumption Sum for 123456789: 1144.4980000000005
2024-03-30 13:42:40.077 DEBUG (MainThread) [custom_components.iec.coordinator] Last Estimated Cost Sum for 123456789: 703.2940209999996
2024-03-30 13:42:40.080 DEBUG (MainThread) [custom_components.iec.coordinator] Last hour fetched for 123456789: 2024-03-28 01:00:00+02:00
2024-03-30 13:42:40.080 DEBUG (MainThread) [custom_components.iec.coordinator] New Consumption Sum for 123456789: 1144.6890000000005
2024-03-30 13:42:40.081 DEBUG (MainThread) [custom_components.iec.coordinator] New Estimated Cost Sum for 123456789: 703.4113904999996
```


In the DB - you see the sum "going back":
|id|created|created_ts|metadata_id|start|start_ts|mean|min|max|last_reset|last_reset_ts|state|sum|
|--|-------|----------|-----------|-----|--------|----|---|---|----------|-------------|-----|---|
|8807||1711673202.5337012|2||1711580400||||||0.191|1144.6890000000005|
|8806||1711673202.5280845|2||1711576800||||||0.47100000000000003|1144.9690000000005|
|8767||1711669588.4814517|2||1711573200||||||0.583|1144.4980000000005|
|8766||1711669588.4797783|2||1711569600||||||2.327|1143.9150000000004|


One thing I suspect is that on both times, it takes the 23:00 (line 3) as the baseline

**Comment by GuyKh** [comment]
I'm releasing this on `0.0.25-beta1` - please leave feedback for if it works (fixes the issue IN THE FUTURE, i.e. don't expect fixing previous negative values)

**Comment by GuyKh** [comment]
Closing as fixed for version `0.0.25`
